src/k2/module/k23si/client/k23si_client.cpp

```diff
@@ -277,8 +280,8 @@ seastar::future<EndResult> K2TxnHandle::end(bool shouldCommit) {
             return _heartbeat_timer.stop().then([this, s=std::move(status)] () {
                 // TODO get min transaction time from TSO client
                 auto time_spent = Clock::now() - _start_time;
```

Comment:
we should check here if there is any need for sleep and only sleep if >0

Reply:
should raise a separate issue for this because this is part of the master branch also.